.drone.yml

```diff
@@ -3,31 +3,91 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
```

Comment:
@romani Please create secrets `drone_aws_access_key_id` and `drone_aws_secret_access_key` with read/write access to the cache S3 bucket.

Reply:
Given there are more drone credentials now (together with [READ_ONLY_TOKEN](https://github.com/checkstyle/checkstyle/blob/master/.drone.yml#L45)) signing ci config could be idea to protect secrets: https://readme.drone.io/signature/. This would prevent somebody to open PR with printenv-like step to obtain secrets, and will require all changes to `.drone.yml` to be done and signed by authorized members. Please let me know if you want to open a new issue for that.

Reply:
Yes, please enhance our security.

Reply:
But with it possible to run update of yml in PR ?

It might force us update yml , resign it, only after it become executable.

Reply:
Sure, we can do that, let me update drone config to simply download and a script to build zip with all dependencies.

PS
I've created a separate ticket for ci config protection (not sure how important is github token) - maybe low priority for now, but something to come back if more integration+secrets will be added to CI/CD: https://github.com/checkstyle/checkstyle/issues/10690

Reply:
I've updated CI to simply download tar archive as requested. Generated file via (196mb):
```
export MAVEN_OPTS="-Dmaven.repo.local=.m2"
mvn dependency:go-offline
tar -cvf checkstyle-maven-cache.tar .m2
```

Reply:
I executed all commands in from CI on empty repository, size of repo become a bit bigger